docs/docs/platform/quickstarts/nextjs.mdx

```diff
@@ -0,0 +1,619 @@
+                className={styles.button}
+              >
+                {isLoading ? <Spinner size="sm" /> : 'Sign in'}
+              </button>
+
+              {isError ? (
+                <p className={styles['error-text']}>{error?.message}</p>
+              ) : null}
+            </form>
+          </>
+        )}
+      </div>
+
+      <p className={styles.text}>
+        No account yet?{' '}
+        <Link href="/sign-up">
+          <a className={styles.link}>Sign up</a>
+        </Link>
+      </p>
+    </div>
+  );
+};
+
+export default SignIn;
+```
+
+#### 3. Sign-out
+
+Finally, to allow the users to sign out from the app, we can use the Nhost
+`useSignOut` hook:
+
+```jsx title="components/Layout.js"
+import { useSignOut } from '@nhost/nextjs';
+
+const Layout = ({ children = null }) => {
+  const { signOut } = useSignOut();
+
+  const menuItems = [
+    //..
+    {
+      label: 'Logout',
+      onClick: signOut,
+      icon: LogoutIcon,
+    },
+  ];
+
+  //...
+};
+```
+
+### Protect routes
+
+Now that we have implemented authentication, we can easily decide who can access
+certain parts of our application.
+
+In our case, we'll only allow authenticated users to have access to the `/` and
+`/profile` routes. All the other users should be redirected to the `/sign-in`
+page if they try to access those routes.
+
+To do so, we can check the authentication status of the current user using the
+Nhost SDK by creating a
+[high-order component](https://reactjs.org/docs/higher-order-components.html):
+
+```jsx title="withAuth.js"
+import { useRouter } from 'next/router';
+import { useAuthenticationStatus } from '@nhost/nextjs';
+import Spinner from './components/Spinner';
+
+export default function withAuth(Component) {
+  return function (props) {
+    const router = useRouter();
+    const { isLoading, isAuthenticated } = useAuthenticationStatus();
+
+    if (isLoading) {
+      return (
+        <div className="h-screen flex items-center justify-center">
+          <Spinner />
+        </div>
+      );
+    }
+
+    if (!isAuthenticated) {
+      router.push('/sign-in');
+      return null;
+    }
+
+    return <Component {...props} />;
+  };
+}
+```
+
+Then, wrap our Next.js pages, `index.js` and `profile.js`, with it:
+
+<Tabs
+defaultValue="index"
+values={[
+  {label: 'index.js', value: 'index'},
+  {label: 'profile.js', value: 'profile'},
+]}>
+<TabItem value="index">
+
+```js
+import withAuth from '../withAuth';
+
+const Home = () => {
+  //...
+};
+
+export default withAuth(Home);
+```
+
+</TabItem>
+<TabItem value="profile">
+
+```js
+import withAuth from '../withAuth';
+
+const Profile = () => {
+  //...
+};
+
+export default withAuth(Profile);
+```
+
+</TabItem>
+</Tabs>
+
+### Retrieve user data
+
+Finally, let's display the information of the authenticated user throughout his
+dashboard to make the app more personalized.
+
+Getting the current authenticated user data is actually quite easy. Indeed, we
```

Comment:
The sentence starting with "Indeed" doesn't make too much sense to me. We are referring to Nhost as a third party here.

Reply:
fixed.